### PR TITLE
Rework `Rack::Response#initialize` documentation to be RDoc format.

### PR DESCRIPTION
**Pull request by ioquatix**
@jeremyevans If you are able to review this and provide feedback that would be great.

### CODE REVIEW

File: lib/rack/response.rb [hunk 1]
```diff
@@ -38,16 +38,28 @@ def header
...
+    #
+    # Otherwise it is expected +body+ conforms to the normal requiremnets of a
+    # Rack repsonse body, typically implementing one of +#each+ (enumerable
+    # body) or +#call+ (streaming body).
+    #
+    # The +status+ defaults to +200+ which is the "OK" HTTP status code. You
+    # can provide any other valid status code.
     #
-    # Providing a body which responds to #to_str is legacy behaviour.
+    # The +headers+ must be a +Hash+ of key-value header pairs which conform to
```
Comment (ioquatix) [inline lib/rack/response.rb:60]: @jeremyevans do I need to write `+Hash+` here?

File: lib/rack/response.rb [hunk 2]
```diff
@@ -38,16 +38,28 @@ def header
       headers
     end
 
+    # :call-seq:
+    #   Response.new(body = nil, status = 200, headers = {}) => response
```
Comment (jeremyevans) [inline lib/rack/response.rb:42]: I personally don't see the value of the call-seq here.

File: lib/rack/response.rb (hunk 1 above)
Comment (jeremyevans) [reply lib/rack/response.rb:60]: I don't think you need to.  You can use `Hash` or `hash`.  You only need to use `+Hash+` if you want to force it displayed as code.  Either way is fine with me.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
I'm OK with merging after the call-seq is removed.

File: lib/rack/response.rb (hunk 1 above)
Comment (ioquatix) [reply lib/rack/response.rb:58]: I tried out different ways and checked how it was formatted, I'm happy with it being formatted as code for now.

File: lib/rack/response.rb (hunk 2 above)
Comment (ioquatix) [reply lib/rack/response.rb]: Okay, removed.

**Review by jeremyevans — APPROVED** [review]

**Pull request merged**